core/src/main/java/org/axonframework/queryhandling/responsetypes/InstanceResponseType.java

```diff
@@ -35,7 +35,8 @@ public InstanceResponseType(Class<R> expectedResponseType) {
      */
     @Override
     public boolean matches(Type responseType) {
-        return isGenericAssignableFrom(responseType) || isAssignableFrom(responseType);
+        Type unwrapped = unwrapIfTypeFuture(responseType);
```

Comment:
Might be nice to add a test case to the `InstanceResponseTypeTest` for when a `@QueryHandler` annotated function returns a `Future`/`CompletableFuture`. Mainly stating this as the `FutureAsResponseTypeToQueryHandlersTest.java` you've added tests the whole working together, whilst I think testing this single unit would be nice as well.